lib/vrl/compiler/src/state.rs

```diff
@@ -28,6 +28,19 @@ pub struct Compiler {
 }
 
 impl Compiler {
+    /// Creates a new compiler that starts with an initial given typedef.
+    /// Currently this is just being used for tests.
```

Comment:
```suggestion
```

I'd leave this off. It's a public API, so people can use it for whatever need they have.